src/client.coffee

```diff
@@ -354,8 +335,81 @@ module.exports = class Client
     request.write query
     request.end()
 
+  ################################################################################################
+  ################################################################################################
+  ################################################################################################
+  ################################################################################################
+  ################################################################################################
+  ################################################################################################
 
   # private methods
+
+  _ensure_hits_exist_in: (options) ->
+    options.usage || options.usage = { hits: 1 }
+
+  _build_query_string: (provider_key, options) ->
```

Comment:
instead of passing _provider_key_ and _options_ as arguments, you would pass only _options_ and use _@provider_key_ as global var like this?
```
  _build_query_string: (options) ->
    if options.service_token
      query = querystring.stringify options
    else
      replacer = (key, value) ->
        if key == 'service_token'
          undefined
        else
          value
      query = querystring.stringify options, replacer   
      query += '&' + querystring.stringify {provider_key: @provider_key}
    return query
````

Would that be OK?